src/Foreign/JavaScript/TH.hs

```diff
@@ -186,6 +186,9 @@ instance MonadAtomicRef m => MonadAtomicRef (WithWebView x m) where
 withWebViewSingleton :: WebView -> (forall x. WebViewSingleton x -> r) -> r
 withWebViewSingleton wv f = f $ WebViewSingleton wv
```

Comment:
That would not work because we would also be hiding the monomorphization from the compiler at the call site of mainWidgetWithHead'.